### ISSUE TITLE
[Bug] Helm Chart secretAnnotations is not properly indented

### ISSUE THREAD

**Issue by jescarri** [mariadb-operator/mariadb-operator#375]
**Documentation**
- [*] I acknowledge that I have read the relevant [documentation](https://github.com/mariadb-operator/mariadb-operator/tree/main/docs).

**Describe the bug**

When deploying the operator via helm + argo-cd the webhook secret is considered part of the helm release because label propagation, an easy fix is to add this annotation to the secret:

```
argocd.argoproj.io/compare-options: IgnoreExtraneous 
```

However the helm chart is indenting 4 spaces vs 6:

https://github.com/mariadb-operator/mariadb-operator/blob/0ad5569f2175a68ad167cd191e97ae9aeebca6e5/deploy/charts/mariadb-operator/templates/webhook-certificate.yaml#L45

When the helm chart is rendered using theve values:

```
fullnameOverride: "mariadb-operator"
logLevel: INFO
ha:
  enabled: false
metrics:
  enabled: true
webhook:
  cert:
    certManager:
      enabled: true
      issuerRef:
        group: cert-manager.io
        kind: ClusterIssuer
        name: selfsigned
    secretAnnotations:
       argocd.argoproj.io/compare-options: IgnoreExtraneous
  serviceMonitor:
    enabled: true

```

The result is:

```
---
# Source: mariadb-operator/templates/webhook-certificate.yaml
apiVersion: cert-manager.io/v1
kind: Certificate
metadata:
  name: mariadb-operator-webhook-cert
  labels:
    helm.sh/chart: mariadb-operator-0.25.0
    app.kubernetes.io/name: mariadb-operator-webhook
    app.kubernetes.io/instance: mariadb-operator
    app.kubernetes.io/version: "v0.0.25"
    app.kubernetes.io/managed-by: Helm
spec:
  commonName: mariadb-operator-webhook.mariadb-operator.svc
  dnsNames:
    - mariadb-operator-webhook.mariadb-operator.svc.cluster.local
    - mariadb-operator-webhook.mariadb-operator.svc
    - mariadb-operator-webhook.mariadb-operator
    - mariadb-operator-webhook
  issuerRef:
    group: cert-manager.io
    kind: ClusterIssuer
    name: selfsigned
  secretName: mariadb-operator-webhook-cert
  secretTemplate:
    labels:
      helm.sh/chart: mariadb-operator-0.25.0
      app.kubernetes.io/name: mariadb-operator-webhook
      app.kubernetes.io/instance: mariadb-operator
      app.kubernetes.io/version: "v0.0.25"
      app.kubernetes.io/managed-by: Helm
    annotations:
    argocd.argoproj.io/compare-options: IgnoreExtraneous

```
Expected result is:

```
...
    annotations:
       argocd.argoproj.io/compare-options: IgnoreExtraneous
```
for code blocks of your kubectl output or YAML files.

**Expected behaviour**

Chart properly indenting extra secret annotations

**Steps to reproduce the bug**

Run helm template with the provided values.

**Comment by jescarri** [comment]
I also think the chart should not add the helm release labels to the certmanager managed service by removing any helm release labels from the secretTemplate, keeping the ability to add extra metadata is nice tho.

```
secretTemplate:
    labels:
      {{- include "mariadb-operator-webhook.labels" . | nindent 6 }}
```

Let me know what you folks think and I can submit a PR to fix both of this issues.

Thanks!

**Comment by mmontes11** [comment]
Hey there @jescarri ! Thanks for reporting this.

> However the helm chart is indenting 4 spaces vs 6:

The indentation is wrong unfortunately, you are right. We need to add render tests to the helm chart for preventing this mistakes in the future.

> I also think the chart should not add the helm release labels to the certmanager managed service by removing any helm release labels from the secretTemplate, keeping the ability to add extra metadata is nice tho.

Agreed, the current helm release labels aren't adding a lot of value to the secret other than unifying them with the rest of resources, happy to remove them. Do you think it is worth supporting ` .Values.webhook.cert.secretLabels` instead?

**Comment by mmontes11** [comment]
This issue might also be out of your interest:
- https://github.com/mariadb-operator/mariadb-operator/issues/285

Maybe even related?

**Comment by jescarri** [comment]
@mmontes11 thanks for your quick response, I haven't seen any need for adding extra labels to the secrets generated by certmanager, but I think may be worth keeping the option open, in the same way the .Values.webhook.cert.secretAnnotations work.

Let me know your thoughts and I will happily add a PR for fixing this.

**Comment by jescarri** [comment]
and @mmontes11 yes, I ran into a problem where argocd was deleting the secret every few seconds and webhook failed.

**Comment by mmontes11** [comment]
Hey @jescarri ! 

Yes, AFAIK cert-manager works based on annotations only and not labels, but we should leave the door open to future integrations, so I think `.Values.webhook.cert.secretLabels` would make sense, but I leave it up to you 😉 

Thanks a lot for your help troubleshooting the webhook problem!

**Comment by jescarri** [comment]
hey @mmontes11 this is the first PR, which removes the helm chart release labels from the secret.

https://github.com/mariadb-operator/mariadb-operator/pull/395

Let me know what you think, and sorry for not submitting this earlier in the week,

**Comment by jescarri** [comment]
Bug has been fixed in #395, I'm closing the issue.

Thanks for the help!
